lambda/transl_comprehension_utils.mli

```diff
@@ -0,0 +1,164 @@
+        "quoted strings", not {fancy|delimited strings|fancy}. *)
+    val string : loc:Location.t -> string -> lambda
+  end
+
+  (** Apply a Lambda function to some Lambda values, at a location; all the
+      other information needed by [Lapply] can be defaulted away and ignored. *)
+  val apply :
+    ?region_close:region_close ->
+    ?mode:alloc_mode ->
+    ?tailcall:tailcall_attribute ->
+    ?inlined:inlined_attribute ->
+    ?specialised:specialise_attribute ->
+    ?probe:probe_desc ->
+    loc:scoped_location ->
+    lambda -> lambda list -> lambda
+
+  (** Nicer OCaml syntax for constructing Lambda ASTs that operate on integers;
+      created by [int_ops], which includes the necessary location in all the
+      operations *)
+  module type Int_ops = sig
+    (** Integer arithmetic *)
+
+    val ( + ) : lambda -> lambda -> lambda
+    val ( - ) : lambda -> lambda -> lambda
+    val ( * ) : lambda -> lambda -> lambda
+    val ( / ) : lambda -> lambda -> lambda
+
+    (** Integer comparisons *)
+
+    val ( = )  : lambda -> lambda -> lambda
+    val ( <> ) : lambda -> lambda -> lambda
+    val ( < )  : lambda -> lambda -> lambda
+    val ( > )  : lambda -> lambda -> lambda
+    val ( <= ) : lambda -> lambda -> lambda
+    val ( >= ) : lambda -> lambda -> lambda
+
+    (** Boolean logical operators *)
+
+    val ( && ) : lambda -> lambda -> lambda
+    val ( || ) : lambda -> lambda -> lambda
+
+    (** Integer literals *)
+
+    val l0 : lambda
+    val l1 : lambda
+    val i  : int -> lambda
+  end
+
+  (** Construct an [Int_ops] module at the given location *)
+  val int_ops : loc:scoped_location -> (module Int_ops)
+
+  (** Expose functions to construct Lambda calls to primitives; some of their
+      arguments have been given labels, but otherwise they mirror the primitives
+      exactly *)
+  module Primitive : sig
+    (** [make_vect ~length ~init] calls the [caml_make_vect] C primitive, which
+        creates an array of the given [length] containing that many copies of
+        the given [init]ial value *)
+    val make_vect :
+      loc:scoped_location -> length:lambda -> init:lambda -> lambda
+
+    (** [make_float_vect len] calls the [caml_make_float_vect] C primitive,
+        which creates an unboxed float array of length [len] whose contents are
+        uninitialized *)
+    val make_float_vect : loc:scoped_location -> lambda -> lambda
+
+    (** [array_append a1 a2] calls the [caml_array_append] C primitive, which
+        creates a new array by appending [a1] and [a2] *)
+    val array_append : loc:scoped_location -> lambda -> lambda -> lambda
+
+    (** [array_sub a ~offset ~length] calls the [caml_array_sub] C primitive,
+        which creates a new subarray corresponding to the subarray of [a]
+        starting at the given [offset] with the given [length] *)
+    val array_sub :
+      loc:scoped_location -> lambda -> offset:lambda -> length:lambda -> lambda
+  end
+end
+
+module Cps_utils : sig
```

Comment:
Dare I say it, but these should probably go in `Misc`?

Reply:
If they go in `Misc`, should I mark them as more generally useful than `transl_comprehension_utils.mli`?  Or perhaps they could go in a non-`Misc` file but with a more general name?